lib/hanami/routes.rb

```diff
@@ -47,5 +49,50 @@ def self.routes
           end
       MSG
     end
+
+    class << self
+      # @api private
+      def build_routes(definitions = self.definitions)
+        return if definitions.empty?
+
+        proc do
+          definitions.each do |(name, args, kwargs, block)|
+            if block
+              public_send(name, *args, **kwargs, &block)
+            else
+              public_send(name, *args, **kwargs)
+            end
+          end
+        end
+      end
+
+      # @api private
+      def definitions
+        @definitions ||= []
+      end
+
+      # @api private
+      def supported_methods
+        @supported_methods ||= Slice::Router.public_instance_methods
+      end
```

Comment:
Could this also be moved below `private`?

Reply:
(Alternatively, could possibly turn into a private constant, making it clearer that this is intended to be static?)